phpmyfaq/src/phpMyFAQ/Bookmark.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * Bookmark handling.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * @package   phpMyFAQ
+ * @author    Jan Harms <model_railroader@gmx-topmail.de>
+ * @copyright 2004-2023 phpMyFAQ Team
+ * @license   https://www.mozilla.org/MPL/2.0/ Mozilla Public License Version 2.0
+ * @link      https://www.phpmyfaq.de
+ * @since     2023-07-19
+ */
+
+namespace phpMyFAQ;
+
+use phpMyFAQ\Configuration;
+use phpMyFAQ\User\CurrentUser;
+
+/**
+ * Class Bookmark
+ *
+ * @package phpMyFAQ
+ */
+class Bookmark
+{
+    /**
+     * CurrenUser object
+     *
+     * @var CurrentUser $user
+     */
+    private CurrentUser $user;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config Configuration object
+     */
+    public function __construct(private Configuration $config)
+    {
+        $this->user = CurrentUser::getFromSession($this->config);
```

Comment:
I would suggest to move the user object to the parameters of the constructor to re-use the user instance